cylc/flow/suite_db_mgr.py

```diff
@@ -576,6 +565,8 @@ def on_restart(self):
         self.check_suite_db_compatibility()
         pri_dao = self.get_pri_dao()
         pri_dao.vacuum()
+        self.n_restart = pri_dao.select_suite_params_restart_count() + 1
+        self.put_suite_params_1(self.KEY_RESTART_COUNT, self.n_restart)
         pri_dao.close()
```

Comment:
Setting this as a property of the suite database manager class feels a bit weird, but I couldn't rely on reading the updated value from the database from the scheduler, because this "put" operation is merely queued and doesn't get executed until after the scheduler has already logged the number of restarts